web/modules/custom/ai_screening_project_track/src/Form/ThresholdsForm.php

```diff
@@ -0,0 +1,136 @@
+    $projectTrackTypes = $this->projectTrackTypeHelper->loadTerms();
+    foreach ($projectTrackTypes as $termId => $projectTrackType) {
+      $projectTrackConfiguration = $this->projectTrackTypeHelper->getConfiguration($projectTrackType);
+
+      $form['project_track'][$termId] = [
+        '#type' => 'fieldset',
+        '#title' => $projectTrackType->label(),
+      ];
+
+      $form['project_track'][$termId]['term_wrapper'] = [
+        '#type' => 'container',
+      ];
+
+      $form['project_track'][$termId]['term_wrapper']['text'] = [
+        '#markup' => '<div class="mb-3 font-bold">' . $projectTrackType->getDescription() . '</div>',
+      ];
+
+      foreach ($projectTrackConfiguration['dimensions'] as $key => $dimension) {
+        $form['project_track'][$termId]['term_wrapper'][$key] = [
+          '#type' => 'container',
+          '#attributes' => ['class' => ['grid', 'gap-4', 'grid-cols-2']],
+        ];
+
+        $form['project_track'][$termId]['term_wrapper'][$key]['text'] = [
+          '#prefix' => '<div>',
+          '#suffix' => '</div>',
+          '#markup' => $dimension,
+        ];
+
+        $form['project_track'][$termId]['term_wrapper'][$key]['dimensions_wrapper'] = [
+          '#type' => 'container',
+          '#attributes' => ['class' => ['grid', 'gap-4', 'grid-cols-2', 'border-bottom']],
+        ];
+
+        $form['project_track'][$termId]['term_wrapper'][$key]['dimensions_wrapper']['approved-' . $termId . '-' . $key] = [
+          '#type' => 'number',
+          '#title' => $this->t('Approved'),
+          '#default_value' => $this->projectTrackTypeHelper->getThreshold($termId, $key, 'approved'),
+          '#min' => 0,
+          '#description' => $this->t('Undecided -> Approved'),
+        ];
+
+        $form['project_track'][$termId]['term_wrapper'][$key]['dimensions_wrapper']['undecided-' . $termId . '-' . $key] = [
+          '#type' => 'number',
+          '#title' => $this->t('Undecided'),
+          '#default_value' => $this->projectTrackTypeHelper->getThreshold($termId, $key, 'undecided'),
+          '#description' => $this->t('Refused -> Undecided'),
+          '#min' => 0,
+        ];
+      }
+    }
+
+    $form['#validate'][] = $this->validateThresholds(...);
+    $form['actions'] = [
+      '#type' => 'actions',
+      'submit' => [
+        '#type' => 'submit',
+        '#value' => $this->t('Save'),
+      ],
+    ];
+
+    return $form;
+  }
+
+  /**
+   * Custom validation for group part of form.
+   */
+  public function validateThresholds(array &$form, FormStateInterface $form_state): void {
+    $userInput = $this->getSubmitted($form_state);
+    foreach ($userInput as $key => $value) {
+      $thresholdKeyArr = explode('-', $key);
+      $thresholds = [
+        'undecided' => $userInput["undecided-{$thresholdKeyArr[1]}-{$thresholdKeyArr[2]}"],
+        'approved' => $userInput["approved-{$thresholdKeyArr[1]}-{$thresholdKeyArr[2]}"],
```

Comment:
Consider adding en `Enum` for these values to give some type safety rather than passing string values around.

Reply:
Changed